File: s6-optional-challenges/frequency_counter/are_there_duplicates/solution.py
def areThereDuplicates(*args):
    unique = {}
    for i in args:
        try:
            unique[i]
        except:
            unique[i] = 1
        else:
            return True
    return False


print(areThereDuplicates(1, 2, 3))  # false
print(areThereDuplicates(1, 2, 2))  # true
print(areThereDuplicates("a", "b", "c", "a"))  # true

File: s6-optional-challenges/frequency_counter/are_there_duplicates/solution.js
/*
Implement a function called, areThereDuplicates which accepts a variable number of arguments, and check whether there are any duplicates among the arguments passed in.
You can solve this using the frequency counter pattern OR the multiple pointers pattern.
*/

const areThereDuplicates = function (...args) {
  const unique = {};
  for (let i of args) {
    if (unique[i]) return true;
    unique[i] = 1;
  }
  return false;
};

console.log(areThereDuplicates(1, 2, 3)); // false
console.log(areThereDuplicates(...[1, 2, 2])); // true
console.log(areThereDuplicates("a", "b", "c", "a")); // true
